### ISSUE TITLE
[GeoMechanicsApplication] When applied load is removed in staged analysis, the effective stress does not update accordingly

### ISSUE THREAD

**Issue by rfaasse** [KratosMultiphysics/Kratos#11978]
**Reproduction steps**
1. Run a single stage where a load is applied on (a part of the) model -> The effective stresses are in accordance with the applied load, this is still correct.
2. Run a second stage where this load is no longer there in the project parameters ->When looking at the effective stress, it seems the applied load of stage 1 is still remembered.

A reproducer case can be found by removing the normal load process from the second stage in GeoMechanicsApplication\tests\test_element_lab\triaxial_comp_6n and seeing that the test doesn't fail (while the effective stress is asserted, and takes into account this applied load).

**Acceptance criteria**
**Given** A stage has run where a load was applied
**When** The user runs a second stage where the aforementioned load is removed
**Then** The effective stress results reflect the new situation without the applied load.

**Comment by rfaasse** [comment]
@avdg81 and myself did an investigation on this issue, these are our findings:

The first thing we checked was if the processes are correct for the staged analysis. When printing a list of the processes during stage 1 and stage 2 of the mentioned triaxial test case, the **_correct_** number of processes is shown for **_both_** stages. This means it is **_not_** the case that processes are remembered in between stages. A positive outcome.

When looking at e.g. the normal load process, what it does in this test is change the solution step value of the NORMAL_CONTACT_STRESS. When looking at the values of this variable after stage 1, we see that the normal load is indeed applied to the top and right side of the domain, as expected, since that's where the normal load was applied in stage 1:
![Image](https://github.com/KratosMultiphysics/Kratos/assets/56549273/c1ccaa17-72c7-4dd3-9d28-231f9c3d6e38)

However, in between stages, we did not find any place where we reset this normal contact stress. Therefore, when we check the NORMAL_CONTACT_STRESS after stage 2, we see the same values, _even though stage 2 does not apply the normal load:_
![Image](https://github.com/KratosMultiphysics/Kratos/assets/56549273/a7e8eaa4-c255-4fc4-ab44-a1433a5aac19)

We think, this is intended behavior: only after explicitly setting the load to 0 again would we remove the load. Whether we agree with this intended behavior or want to spend effort to try and change this, we can discuss.

In summary, we found no major issues in the workflow, but found some behaviour that we need to document, since it's not trivial and discuss whether this is the state we would like to be in. In my opinion this means the urgency of this bug can be lowered.

FYI @mcgicjn2 @WPK4FEM

**Comment by rfaasse** [comment]
After feedback, @WPK4FEM and myself had another look at this issue and it turns out that if you remove the sub model parts which are related to in this case the normal load process from the `processes_sub_model_part_list`, the NORMAL_CONTACT_STRESS and effective stress are as expected. 

This means that the load will only remain, if the modelpart (which has the load-condition attached to it) is specified again in a later stage, even though the process is not there.

Since this means that with correct and clean input files, the workflow works as expected, I would propose to close this issue, since it's not really the bug we thought it was.

However, we had a discussion about this and would like to make the initialization of the sub model parts more robust, i.e. only specify the relevant model parts once and not both in the processes and in the `processes_sub_model_part_list`. I created a draft issue for this and put it on the backlog to be prioritized.

FYI @mcgicjn2 @indigocoral
